datajoint/external.py

```diff
@@ -79,38 +79,49 @@ def connect(self):
 
     def stat(self, rpath=None):
         """
-        check if a file exists in the bucket
+        Check if a file exists in the bucket.
+
+        :param rpath: remote path within bucket
         """
         try:
             self.connect()
             self._s3.Object(self._bucket, rpath).load()
         except ClientError as e:
-            if e.response['Error']['Code'] == "404":
-                return False
-            else:
-                raise DataJointError('error checking remote file')
+            if e.response['Error']['Code'] != "404":
+                raise DataJointError('Error checking remote file', str(rpath),
```

Comment:
Throughout the module, I think it's better to use formatted strings than string sums.  For example, this line should be
```python
raise DataJointError('Error checking remote file {p} ({e})'.format(p=rpath, e=e))
``` 
It's more explicit and efficient.

Reply:
switched.. and agreed (assuming you are correct) for the general case of 'how to manage strings in the codebase'.. that said if we are worried about string processing efficiency within exception constructors we have biggerr problems on our hands :D

Reply:
disagree this is more explicit however.. but in any event, /usr/local/bin/go with-the-flow.go